apps/web/src/lib/costs/get-national-average-cost.tsx

```diff
@@ -0,0 +1,61 @@
+function validateInputs(userData, days) {
+    if (typeof userData !== "object" || userData === null) {
+        throw new Error("Invalid input: userData must be an object.");
+    }
+    if (!Number.isInteger(days) || days < 1) {
+        throw new Error("Invalid input: days must be a positive integer.");
+    }
+    if (!Object.prototype.hasOwnProperty.call(userData, "household")) {
+        throw new Error("Missing household information in userData.");
+    }
+    if (!Number.isInteger(userData.household) || userData.household < 1 || userData.household > 8) {
+        throw new Error("Invalid household size. Must be between 1 and 8.");
+    }
+}
+
+export function getNationalAverageCost(userDatas, days) {
+    if (!Array.isArray(userDatas) || userDatas.length === 0) {
+        console.error("Invalid input: userData must be a non-empty array.");
+        return null;
+    }
+
+    const userData = userDatas[1];
+
+    try {
+        validateInputs(userData, days);
```

Comment:
Muss man sowas extra nochmal prüfen?

Reply:
Hatte irgendwie erst Probleme mit den Inputs deswegen hab ich das hinzugefügt aber kann man auch weglassen

Reply:
Wenn das geht würde ich das weglassen. Und darüber bei const userData = userDatas[1]; nicht den zweiten, sondern den letzten Wert nehmen. DA hatte ich schon mit Daniel drüber gesprochen

Reply:
Hab die Prüfung jetzt drin gelassen, weil es ja nicht schadet. Habe es aber zum letzten Wert geändert.

Reply:
Ja ist halt unnötiger Code an der Stelle, weil die Prüfung eigentlich gemacht wird, nachdem die Daten geladen werden. Aber ich kenn das selber, dass eine Variable dann rot angestrichen wird, wenn man die nicht prüft😄 deswegen passt schon

Reply:
Eine Exception manuell werfen und das direkt in einem Try-Catch abfangen, ist meiner Meinung nach eigentlich bad practice.
Besser fände ich hier eigentlich sowas:
```suggestion
        if (typeof userData !== "object" || userData === null) {
          console.error("Invalid input: userData must be an object.");
          return null;
      }
      if (!Number.isInteger(days) || days < 1) {
          console.error("Invalid input: days must be a positive integer.");
          return null;
      }
      if (!Object.prototype.hasOwnProperty.call(userData, "household")) {
          console.error("Missing household information in userData.");
          return null;
      }
      if (!Number.isInteger(userData.household) || userData.household < 1 || userData.household > 8) {
          console.error("Invalid household size. Must be between 1 and 8.");
          return null;
      }
      
      // restlicher Code
```

Aber wäre jetzt nicht so wichtig. Wenn du nichts weiter sonst anpassen musst, lass das meinetwegen einfach so